### PR TITLE
8328087: Automate javax/swing/JTable/TAB/TAB.java applet test

### PR DESCRIPTION
**Pull request by kumarabhi006**
Conversion of manual applet test to automated test. CI testing is green. Link attached to JBS.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8328087](https://bugs.openjdk.org/browse/JDK-8328087): Automate javax/swing/JTable/TAB/TAB.java applet test (**Bug** - P4)

### Reviewers
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - **Reviewer**) ⚠️ Review applies to [0a6e5acb](https://git.openjdk.org/jdk/pull/18274/files/0a6e5acb3299ea84ee273647048f95072633bde1)
 * [Tejesh R](https://openjdk.org/census#tr) (@TejeshR13 - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18274/head:pull/18274` \
`$ git checkout pull/18274`

Update a local copy of the PR: \
`$ git checkout pull/18274` \
`$ git pull https://git.openjdk.org/jdk.git pull/18274/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18274`

View PR using the GUI difftool: \
`$ git pr show -t 18274`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18274.diff">https://git.openjdk.org/jdk/pull/18274.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18274#issuecomment-1994414354)

### CODE REVIEW

File: test/jdk/javax/swing/JTable/TAB.java [hunk 1]
```diff
@@ -40,18 +44,82 @@
...
- * @author milne
- * @run applet/manual=yesno TAB.html
+ * @key headful
+ * @summary Verify focus changes correctly when tab is pressed while editing
+ *          a JTextField in a JTable
+ * @run main TAB
  */
-public class TAB extends JApplet
+
+public class TAB
```
Comment (honkar-jdk) [inline test/jdk/javax/swing/JTable/TAB.java:56]: Class name is in all-caps, `Tab` would be more appropriate.

**Review by honkar-jdk — CHANGES_REQUESTED** [review]

**Comment by kumarabhi006** [comment]
@honkar-jdk Updated as per last review. Please take a look.

File: test/jdk/javax/swing/JTable/Tab.java [hunk 2]
```diff
@@ -40,43 +44,107 @@
...
- * @run applet/manual=yesno TAB.html
+ * @key headful
+ * @summary Verify focus changes correctly when tab is pressed while editing
+ *          a JTextField in a JTable
+ * @run main Tab
  */
-public class TAB extends JApplet
+
+public class Tab
 {
```
Comment (honkar-jdk) [inline test/jdk/javax/swing/JTable/Tab.java:57]: Minor nit: Opening brace placement

**Review by honkar-jdk — APPROVED** [review]

**Review by TejeshR13 — COMMENTED** [review]
Looks good to me. Does this work fine in other L&F (Like Aqua), because you have explicitly mentioned as Metal right?

**Comment by kumarabhi006** [comment]
> Looks good to me. Does this work fine in other L&F (Like Aqua), because you have explicitly mentioned as Metal right?

The older version of applet test also checks for MetalLookAndFeel. Since the test is just converted to main manual, didn't check explicitly on Aqua or other LAFs.

**Review by TejeshR13 — APPROVED** [review]

**Pull request closed without merging**